.github/workflows/coverage.yml

```
@@ -0,0 +1,149 @@
+      - name: Extract coverage percentage
+        id: coverage
+        run: |
+          chmod +x scripts/extract-coverage.sh
+          COVERAGE=$(./scripts/extract-coverage.sh coverage/lcov.info)
+          echo "percentage=$COVERAGE" >> $GITHUB_OUTPUT
+          echo "Current coverage: $COVERAGE%"
+
+      - name: Get master branch coverage (for PRs)
+        if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
+        id: master_coverage
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2086:info:4:34: Double quote to prevent globbing and word splitting

(shellcheck)